src/dds_manager.cpp

```diff
@@ -953,9 +962,29 @@ bool DDSManager::replaceFilter(const std::string& topicName,
         dynamic_cast<DDS::ContentFilteredTopic*>
         (dataReader->get_topicdescription());
 
+    // We have to destroy the current data reader before building a new one
+    // The first step is to delete the contained entities (deletes all the ReadConditions and QueryConditions)
+    DDS::ReturnCode_t return_code = dataReader->delete_contained_entities();
+    if (return_code != DDS::RETCODE_OK)
+    {
+        std::cerr << "dataReader failed on delete_contained_entities, return_code:  " << return_code << std::endl;
+        return false;
+    }
+
+    // Now delete the reader
+    return_code = subscriber->delete_datareader(dataReader);
```

Comment:
```suggestion
    return_code = subscriber->delete_datareader(dataReader);
    dataReader = nullptr;
```

Since dataReader is reference counted, it's a good practice to drop the reference here and make sure nothing else is using it

Reply:
Done